src/codegen.rs

```diff
@@ -84,6 +87,40 @@ pub struct GeneratedModule<'ink> {
 type MainFunction<T, U> = unsafe extern "C" fn(*mut T) -> U;
 type MainEmptyFunction<U> = unsafe extern "C" fn() -> U;
 
+pub fn read_got_layout(location: &str, format: ConfigFormat) -> Result<HashMap<String, u64>, Diagnostic> {
+    if !Path::new(location).is_file() {
+        // Assume if the file doesn't exist that there is no existing GOT layout yet. write_got_layout will handle
+        // creating our file when we want to.
+        return Ok(HashMap::new());
+    }
+
+    let s =
+        read_to_string(location).map_err(|_| Diagnostic::new("GOT layout could not be read from file"))?;
```

Comment:
I think it is clearer to keep the module when calling free-standing public functions like this one, so I would rather see this:

```suggestion
        fs::read_to_string(location).map_err(|_| Diagnostic::new("GOT layout could not be read from file"))?;
```

and remove the import for `fs::{read_to_string, write}` in favor of an import of `fs`. this is definitely a nit so feel free to ignore, just easier for me to read the code

Reply:
Agreed!